### PR TITLE
Adding support for existing cluster

### PR DESCRIPTION
**Pull request by dunefro**
1. Add support for existing cluster using variable `use_existing_cluster`.
2. Following variables are introduced
```bash
use_existing_cluster
existing_cluster_node_role_arn
existing_cluster_node_security_group_id
existing_cluster_oidc_issuer_arn
existing_cluster_oidc_issuer_url
```
3. moved the eks and eks-addons to a count block. This will cause a drift which should be checked in under the `migration-guide.md`
4. Introduced a scripts folder to ensure better practices for upgrades or any additional use.

### CODE REVIEW

File: output.tf [hunk 1]
```diff
@@ -3,50 +3,55 @@
 ################################################################################
 # Cluster
 ################################################################################
+output "use_existing_cluster" {
+  description = "Flag to check if you are using an alreday existing cluster"
```
Comment (DeeAjayi) [inline output.tf]: description = "Flag to check if you are using an already existing cluster"

**Review by DeeAjayi — CHANGES_REQUESTED** [review]

File: output.tf (hunk 1 above)
Comment (dunefro) [reply output.tf]: nice catch

**Review by DeeAjayi — DISMISSED** [review]

File: locals.tf [hunk 2]
```diff
@@ -104,4 +104,35 @@ locals {
...
+  // tflint-ignore: terraform_map_duplicate_keys
+  output_eks_managed_node_groups = tomap({
+    "initial" = {
+      autoscaling_group_schedule_arns = {}
+      iam_role_arn                    = var.existing_cluster_node_role_arn
+      iam_role_unique_id              = {}
+      launch_template_arn             = ""
+      launch_template_id              = ""
+      // tflint-ignore: terraform_map_duplicate_keys
+      autoscaling_group_schedule_arns = {}
```
Comment (shubhamrai1993) [inline locals.tf]: why are there duplicate keys?

File: scripts/migration-script-7.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
```
Comment (shubhamrai1993) [inline scripts/migration-script-7.sh:1]: why is it called script 7

Comment (dunefro) [reply scripts/migration-script-7.sh:1]: It is migration to module version 7

File: locals.tf [hunk 4]
```diff
@@ -104,4 +104,27 @@ locals {
   } : {})
 
   karpenter_profile_name = "${var.cluster_name}-karpenter"
+
+  // this is used when use_existing_cluster is set to true, so that we don't have to modify eks_managed_node_groups
+  // tflint-ignore: terraform_map_duplicate_keys
```
Comment (shubhamrai1993) [inline locals.tf]: Is this still needed?

File: scripts/migration-script-7.sh (hunk 3 above)
Comment (shubhamrai1993) [reply scripts/migration-script-7.sh:1]: Lets use the complete version identified

**Review by shubhamrai1993 — DISMISSED** [review]

File: locals.tf (hunk 4 above)
Comment (dunefro) [reply locals.tf]: Removed, this doesn't work anyways

**Review by shubhamrai1993 — APPROVED** [review]

**Pull request merged**